torch/_inductor/kernel/flex_attention.py

```diff
@@ -1768,16 +1899,42 @@ def flex_attention_backward(*args, **kwargs):
     fw_subgraph_buffer = build_subgraph_buffer(
         fwd_placeholder_inps + list(score_mod_other_buffers), fw_graph
     )
+    assert isinstance(fw_subgraph_buffer, SubgraphOutput)
+    fw_subgraph_buffer = fw_subgraph_buffer.compute_buffer
 
     joint_placeholder_inps = fwd_placeholder_inps + [
         create_placeholder("grad_score_mod", dtype, device)
     ]
     # Sometimes we have weird unused nodes here
     joint_graph.graph_module.graph.eliminate_dead_code()
-    joint_subgraph_buffer, *_ = build_subgraph_buffer(
-        joint_placeholder_inps + list(score_mod_other_buffers), joint_graph
+
+    # It is hard to raise nice errors for some joint graphs during subgraph lowering
+    # This lets us do some checks before attempting to lower
+    validate_joint_graph(joint_graph.graph_module.graph)
+
+    all_joint_outputs = build_subgraph_buffer(
+        joint_placeholder_inps + list(score_mod_other_buffers),
+        joint_graph,
+    )
+    assert isinstance(all_joint_outputs, List)
+    assert (
+        all_joint_outputs[0] is not None
+    ), "joint_subgraph_buffer is None this is a bug!"
+    joint_subgraph_buffer = all_joint_outputs[0].compute_buffer
+
+    score_mod_other_buffer_grads = all_joint_outputs[len(joint_placeholder_inps) - 1 :]
+
+    score_mod_other_buffer_grads_compute = [
```

Comment:
I used tree map sometimes but nother others cause it splats inner tuples but should probs comment